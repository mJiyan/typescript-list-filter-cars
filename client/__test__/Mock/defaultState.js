const defaultState = {
    make: 'Ford',
    makes: [
        "ALFA ROMEO",
        "AUDI",
        "AUSTIN",
        "BARKAS",
        "BMW",
        "CADILLAC",
        "CHEVROLET",
        "CHRYSLER",
        "CITROEN",
        "CORVETTE",
        "DACIA",
        "DAEWOO",
        "DAF",
        "DAIHATSU",
        "DODGE",
        "FIAT",
        "FORD",
        "HDPIC",
        "HONDA",
        "HYUNDAI",
        "INFINITI",
        "ISUZU",
        "IVECO",
        "JAGUAR",
        "JEEP",
        "KIA",
        "LADA",
        "LANCIA",
        "LAND ROVER",
        "LANDWIND",
        "LDV",
        "LEXUS",
        "MAZDA",
        "MERCEDES-BENZ",
        "MG",
        "MINI (BMW)",
        "MINI (ROVER)",
        "MITSUBISHI",
        "NISSAN",
        "OPEL",
        "PEUGEOT",
        "PONTIAC",
        "PORSCHE",
        "PROTON",
        "RENAULT",
        "ROVER",
        "SAAB",
        "SEAT",
        "SKODA",
        "SMART",
        "SSANGYONG",
        "SUBARU",
        "SUZUKI",
        "TALBOT",
        "TATA",
        "TESLA",
        "TOYOTA",
        "VOLVO",
        "VW"
    ],
    models: [
        "B-MAX",
        "C-Max",
        "Cougar",
        "Econovan",
        "ECOSPORT",
        "Edge",
        "Escort",
        "Explorer",
        "Fiesta",
        "Fiesta Courier",
        "Focus",
        "Focus C-Max",
        "Fusion",
        "Galaxy",
        "KA",
        "Kuga",
        "Maverick",
        "Mondeo",
        "Mustang",
        "Orion",
        "Probe",
        "Puma",
        "Scorpio",
        "Sierra",
        "S-Max",
        "Streetka",
        "Tourneo Connect",
        "Tourneo Courier",
        "Transit",
        "Transit Connect",
        "Transit Courier",
        "Transit Custom",
        "Windstar"
    ],
    vehicles: [
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 200,
            "enginePowerKW": 147,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1084
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 86,
            "enginePowerKW": 63,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 125,
            "enginePowerKW": 92,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 65,
            "enginePowerKW": 48,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 182,
            "enginePowerKW": 134,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 135,
            "enginePowerKW": 99,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 150,
            "enginePowerKW": 110,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1999
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1271
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 49,
            "enginePowerKW": 36,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1567
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 82,
            "enginePowerKW": 60,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1242
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 120,
            "enginePowerKW": 88,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 90,
            "enginePowerKW": 66,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 140,
            "enginePowerKW": 103,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 998
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 88,
            "enginePowerKW": 65,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1597
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 45,
            "enginePowerKW": 33,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 935
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 103,
            "enginePowerKW": 76,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1598
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1087
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1753
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 73,
            "enginePowerKW": 54,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1119
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 50,
            "enginePowerKW": 37,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 105,
            "enginePowerKW": 77,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 131,
            "enginePowerKW": 96,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1796
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 54,
            "enginePowerKW": 40,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1097
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 71,
            "enginePowerKW": 52,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1391
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 69,
            "enginePowerKW": 51,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 60,
            "enginePowerKW": 44,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1299
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1368
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 80,
            "enginePowerKW": 59,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 75,
            "enginePowerKW": 55,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1560
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 95,
            "enginePowerKW": 70,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1499
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 97,
            "enginePowerKW": 71,
            "fuelType": "Gas",
            "bodyType": "Limousine",
            "engineCapacity": 1388
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 68,
            "enginePowerKW": 50,
            "fuelType": "Diesel",
            "bodyType": "Limousine",
            "engineCapacity": 1399
        },
        {
            "make": "FORD",
            "model": "Fiesta",
            "enginePowerPS": 101,
            "enginePowerKW": 74,
            "fuelType": "Benzin",
            "bodyType": "Limousine",
            "engineCapacity": 1596
        }
    ]
}

export default defaultState;