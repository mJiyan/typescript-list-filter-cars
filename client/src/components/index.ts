import Header from './Header';
import Card from './Card';
import Filter from './Filter';
import Search from './Search';
import MakeList from './MakeList';
import ModelList from './ModelList';
import CardList from './CardList';


export { Header, Card, Filter, Search, MakeList, ModelList, CardList }