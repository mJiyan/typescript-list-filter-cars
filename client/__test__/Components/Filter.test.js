import React from 'react';
import { Filter } from '../../src/components';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import { Router, Route, Switch } from 'react-router-dom';
import { createMemoryHistory } from 'history'

import defaultState from '../Mock/defaultState';
import toJson from 'enzyme-to-json'; //added this line
import 'jsdom-global/register'; //at the top of file , even  , before importing react

const mockStore = configureMockStore([thunk]);
const store = mockStore(defaultState);
const history = createMemoryHistory()

const wrapper = renderer.create(
    <Provider store={store}>
        <Router history={history}>
            <Switch>
                <Filter />
            </Switch>
        </Router>
    </Provider>,
);


const wrapperMount = mount(
    <Provider store={store}>
        <Router history={history}>
            <Switch>
                <Filter />
            </Switch>
        </Router>
    </Provider>,
);

describe('rendering filter components', () => {
    test('it should match with the snapshot', () => {
        expect(wrapper.toJSON()).toMatchSnapshot();
    });

    test('it should find the search box', () => {
        const searchInput = wrapperMount.find('[data-testid="search-input"]');
        expect(searchInput.length).toEqual(1);
    });



});