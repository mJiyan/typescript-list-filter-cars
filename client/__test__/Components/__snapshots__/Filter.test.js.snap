// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering filter components it should match with the snapshot 1`] = `
<div
  className="flex 
                    xl:flex-row lg:flex-row  md:flex-row sm:flex-col xs:flex-col 
                    justify-between 
                    xl:px-20 lg:px-20 md:px-20 sm:px-3 xs:px-3
                    xl:pt-8 lg:pt-8 md:pt-8 sm:pt-5 xs:pt-5
                    xl:h-20 lg:h-20 md:h-20 sm:h-full xs:h-full"
>
  <div
    className="xl:w-1/6 lg:w-1/6 md:w-1/6 sm:w-2/3 xs:w-2/3 
      xl:h-full lg:h-full md:h-full sm:h-12 xs:h-12
      xl:my-0 lg:my-0 md:my-0 sm:my-10 xs:my-10"
  >
    <button
      className="flex items-center justify-between border-none shadow rounded-md w-full h-full text-sm px-6 bg-WhiteDark dark:bg-DarkBlue "
      data-testid="toggle-make-list"
      onClick={[Function]}
    >
      Select a Make
      <svg
        className="h-5 w-5"
        fill="currentColor"
        viewBox="0 0 20 20"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
  <div
    className="flex items-center 
      xl:w-2/6 lg:w-2/6 md:w-2/6 sm:w-full xs:w-full 
      xl:h-full lg:h-full md:h-full sm:h-12 xs:h-12"
  >
    <svg
      className="h-5 w-5 absolute ml-6"
      fill="currentColor"
      viewBox="0 0 20 20"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        clipRule="evenodd"
        d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
        fillRule="evenodd"
      />
    </svg>
    <input
      className="border-none rounded-md text-sm bg-WhiteDark dark:bg-DarkBlue shadow w-full h-full px-16 placeholder-VeryDarkBlueLight dark:placeholder-WhiteDark"
      data-testid="search-input"
      onChange={[Function]}
      placeholder="Search for a vehicle fuelType..."
    />
  </div>
</div>
`;
